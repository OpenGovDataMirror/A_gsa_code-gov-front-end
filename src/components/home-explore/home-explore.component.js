import React, { Fragment } from 'react'
import CustomLink from 'components/custom-link'
import { startsWith } from '@code.gov/cautious'

function LinkButtons({ name, className, url }) {
  if (startsWith(url, 'http') || startsWith(url, 'mailto')) {
    return (
      <a href={url} className={className} target="_blank" rel="noopener noreferrer">
        {name}
      </a>
    )
  }
  return (
    <CustomLink to={url} className={className}>
      {name}
    </CustomLink>
  )
}

export default class HomeExplore extends React.Component {
  render() {
    return (
      <div className="grid-row">
        <section id="explore" className="explore usa-section">
          <div className="explore-actions grid-row text-center tablet:border-bottom-2px tablet:border-top-2px tablet:padding-top-205 padding-top-3 tablet:padding-bottom-205 padding-bottom-0">
            {this.props.exploreItems &&
              this.props.exploreItems.map(item => (
                <div
                  className="explore-action grid-row flex-column flex-justify  padding-right-2 padding-left-2 tablet:grid-col-4 tablet:border-right-2px margin-bottom-5 tablet:margin-bottom-0"
                  key={item.title}
                >
                  <div>
                    <h2 className="text-primary">{item.title}</h2>
                    <p dangerouslySetInnerHTML={{ __html: item.description }} />
                  </div>
                  <div className="buttons">
                    {item.links.map(link => (
                      <LinkButtons
                        key={link.name}
                        className="usa-button font-body-sm tablet-lg:padding-x-5 line-height-sans-2"
                        name={link.name}
                        url={link.url}
                      />
                    ))}
                  </div>
                </div>
              ))}
          </div>
        </section>
      </div>
    )
  }
}
