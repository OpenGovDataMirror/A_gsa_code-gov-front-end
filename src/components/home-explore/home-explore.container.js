import { connect } from 'react-redux'
import { getConfigValue } from 'utils/other'
import HomeExploreComponent from './home-explore.component'

export const mapStateToProps = () => ({
  exploreItems: getConfigValue('content.home.explore')
})

export default connect(mapStateToProps)(HomeExploreComponent)
