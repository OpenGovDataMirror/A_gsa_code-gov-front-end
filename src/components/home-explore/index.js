import HomeExploreContainer from './home-explore.container'

export default HomeExploreContainer
