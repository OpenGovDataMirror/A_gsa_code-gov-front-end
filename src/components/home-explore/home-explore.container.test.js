import { mapStateToProps } from 'components/home-explore/home-explore.container'

describe('containers - HomeExplore', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps()).toMatchSnapshot()
    })
  })
})
