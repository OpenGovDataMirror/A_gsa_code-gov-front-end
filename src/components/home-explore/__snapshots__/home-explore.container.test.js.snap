// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomeExplore mapStateToProps should return the correct properties 1`] = `
Object {
  "exploreItems": Array [
    Object {
      "description": "The Code.gov website is developed publicly on GitHub. Learn how to <a href='https://github.com/GSA/code-gov/blob/master/README.md' target='_blank' rel='noopener noreferrer'>contribute here</a>. Help improve America's Code by exploring projects.",
      "links": Array [
        Object {
          "name": "Explore Projects",
          "url": "/agencies",
        },
      ],
      "title": "Developers",
    },
    Object {
      "description": "Have questions or feedback? <a href='https://github.com/GSA/code-gov-front-end/issues/new/choose' target='_blank' rel='noopener noreferrer'>Open an issue</a> on our open source repository.<br><br>Want to receive our quarterly newsletter?",
      "links": Array [
        Object {
          "name": "Sign Up Here",
          "url": "https://public.govdelivery.com/accounts/USGSATTS/subscriber/new?topic_id=USGSATTS_5",
        },
      ],
      "title": "Connect with Us",
    },
    Object {
      "description": "Federal agency partners use Code.gov to share usable open source code, promote open source projects, and track compliance with federal open source policy.",
      "links": Array [
        Object {
          "name": "Agency Compliance",
          "url": "/agency-compliance/compliance/dashboard",
        },
      ],
      "title": "Agency Partners",
    },
  ],
}
`;
